bpf/lib/encrypt.h

```diff
@@ -284,8 +288,109 @@ encrypt_overlay_and_redirect(struct __ctx_buff *ctx)
 	return ret;
 }
 #endif /* ENABLE_ENCRYPTED_OVERLAY */
+#elif defined(ENABLE_WIREGUARD)
+
+/* Returns CTX_ACT_REDIRECT in case of a cilium-managed WireGuard packet.
+ * Differently than do_encrypt() for IPSec, WireGuard does not use the ctx->mark
+ * to indicate whether a packet needs decryption. Therefore, we can just return
+ * CTX_ACT_REDIRECT to the caller, or any other value in case of non-WireGuard packet.
+ * The conditions sequentially checked in order to establish that a WireGuard packet is
+ * valid and should be traced with TRACE_REASON_ENCRYPT are as follows:
+ *
+ * - UDP packet
+ * - L4 dport == WG_PORT
+ * - L4 sport == dport
+ * - Valid node ID
+ */
+static __always_inline int
+do_decrypt(struct __ctx_buff *ctx, __u16 proto)
```

Comment:
I see your point: ipsec has a `do_decrypt` so you may think wireguard should have a similar one. But I feel you just need a  `ctx_is_wireguard(__ctx_buff) bool`.

Reply:
Agree, still would make use of the `__u16 proto` already computed.  For which file this new function would be a better fit rather than `encrypt.h`?